packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -186,7 +189,14 @@ const Dashboard: FC = () => {
 
     const { tiles: savedTiles } = dashboard || {};
     useEffect(() => {
+        // TODO: The logic in this useEffect isn't right. It's checking if
+        // there are saved tiles, then checking for unsaved tiles,
+        // then replacing the saved tiles with the unsaved ones if they exist.
         if (savedTiles) {
+            // TODO: maybe this should move in the future, but it makes
+            // some sense here since this useEffect is essentially handling
+            // sessions storage
+            clearIsEditingDashboardChart();
             const unsavedDashboardTilesRaw = sessionStorage.getItem(
                 'unsavedDashboardTiles',
             );
```

Comment:
Doesn't line 199 delete the storage data that is then requested in the next line ?

Reply:
No. The pattern is a little confusing, but I'm working with what was already in place. There are essentially 2 pieces of information in storage 1) Whether or not we are editing a chart in a dashboard and which dashboard, used by the header and the explores page, 2) The data about the unsaved charts that are being edited. When we land on the dashboard page we need to clear #1 since we are no longer editing a chart in a dashboard. Then we pick up number #2 to add the unsaved charts to the dashboard. There is a separate clear for clearing the local storage. So we have:

- `clearIsEditingDashboardChart` -> to indicate we are no longer editing a chart in a dashboard
- `clearDashboardStorage` -> clears out all of the dashboard storage

I'm open to suggestions to make this more obvious. I did it this way partly to make it work with what's already there. I can keep my eye on cleaning up this pattern a bit. As I say in the comment, I think this useEffect is a bit wrong anyway -- it's possible the hook should just provide some way to get the stored charts and clear state as one operation. Thoughts?

Reply:
thanks for explaining. I was mixing up the 2 clear functions.